# Linear Regression in English

Linear regression is about finding relationships between numeric values. We divide data into independent variables (that we know value a-prior) and dependent variable (that we want to predict).

In Linear Regression, we assume the relationship is linear, which can be decribed as a straight line going through all observations, so the sum of distance from all observations to the straight line is minimal (when the straight line becomes the best approximation for the observations).

P value is the chance of NUll hypothesis of being rejected by mistake (of data sampling).

R Squared is to measure the squared distance of fitted regression line divided by the squared distance of a simple average. The smaller proportion the better (a better fit)


![Image of Yaktocat](http://i63.tinypic.com/2ut6l8g.png)

# Small multiples - 

Project: http://www.theguardian.com/news/datablog/interactive/2013/oct/27/interactive-safest-time-to-drive

Purpose: To visualise the pattern of car accidents by month, and by time in the day.
What I found:![Image of Yaktocat]( http://i66.tinypic.com/oqzj90.png ) The upper half seems have too much animation and not clear about what meaning to convey; the bottom part is straight forward, which shows monthly % of annual car acci.dents




# NULL hypothesis: 

Media (TV, Radio, newspaper etc) budget is not relevant to Prduct Sales

Conclusion: TV and Radio are both significantly  related to the Sales. Newspaper is not correlated to Sales (and can be removed from the regression model)



# KNN
The algorithm is the same, for regression the average result of y is used as prediction. For classification the most frequent category by voting. For regression, we can use average of N neighbours as the predicted value.

# 3 ways to select feature:
1. Forward selection - start with NULL model, then add feature brings lowest RSS, and repeat the process until
2. Backward selection
3. Mixed selection
2. check effeciency
3. add to model to see if accuracy changed

Exercise 3.2:


In [1]:
def reg(GPA,IQ,Gender):
 b=[50,20,0.07,35,0.01,-10]

 a=b[0]+GPA*b[1]+IQ*b[2]+Gender*b[3]+GPA*IQ*b[4]+GPA*Gender*b[5]


 print (a)
 return a


if __name__ == '__main__':  
 reg(110,4,1)

1189.68


Exercise:

Statement IIIFor fixed value of GPA/IQ, male with high enough GPA earns more. 

Salary of Female/IQ110/GPA4.0 is 1190

False, as 0.01 / 0.07 are at the same scale, which means if GPA is high enough, the interaction of GPA/IQ will have similar impact as IQ alone.

Regression Techniques that I may use in my data project:

linear regression, logistic regression. Both can be used to estimate the value of missing data. This may take some time but should bring better result than using average etc.

